upsilonconf/config.py

```diff
@@ -404,17 +404,56 @@ def _replace_in_keys(
     config : Mapping
         The configuration object (or dictionary) to be modified.
     key_modifiers : dict
-        The dictionary with the replacements.
+        The dictionary with the replacements: All keys strings are replaced
+        with the corresponding values from the `key_modifiers` dictionary.
 
     Returns
     -------
     dict
         A dictionary with the modified keys.
 
+    Replacement strategy
+    --------------------
+    1. The replacement should be zealous, meaning that
+       `key_modifiers={'  ': '_', ' ': '-'}` should replace a single spaces
+       with a hyphen and two spaces with an underscore (and not two hypens).
+    2. The replacement should not be recursive, meaning that
+       `key_modifiers={' ': '-', '-': '_'}` should replace a hyphen with an
+       underscore and a space with a hyphen (and not an underscore).
     """
+    if not key_modifiers:
+        return dict(config)
```

Comment:
When I run
```python3
python3 -m unittest
```
I don't get any error at all.

BTW the `return dict(config)` is a verbatim copy from your commit and is needed, because the tests require a `dict` to be returned. But it is true, that the nested objects are not converted to dictionaries. Should this be done recursively?

Reply:
> I don't get any error at all.

Then I probably have to take another look at the tests.

> BTW the `return dict(config)` is a verbatim copy

Yes, but I had it at the end (after the recursion).

Reply:
But after the recursion the conversion `dict(config)` it is not needed, because `__replace_in_keys()` _always_ returns a `dict`?? But if there are no replacements, `_replace_in_keys()` does not dive into the values recursively.

Also your tests contradict the returning type of `_replace_in_keys() -> Mapping[str, Any]`?

Do you want `_replace_in_keys()` to convert also nested mappings to dictionaries? That can be done; but to be honest, I cannot even see the reasoning behind converting the top-level mapping to a dictionary, because ‘a mapping looks like a dictionary and quacks like a dictionary’, so why bother?

You might have different views …

Reply:
I fixed that test. 

You are right that the test contradicts the return type. I should probably fix that, but I'll probably do that another time.

You are definitely right about duck typing, but I would like to get consistent behaviour. Concretely: if configs are transformed to dictionaries when `key_modifiers` is specified, then I would prefer to have the same behaviour if it is not specified. Moreover, this makes the method also have a purpose if `key_modifiers` is not specified.